### DISCUSSION TITLE
Error for unknow modification

### DISCUSSION THREAD

**Discussion by grazzilybear** [vdemichev/DiaNN#661] [General]
We are using Fragpipe to generate a library but we used our chemical tag on C (399.19418) as fix modification.

When we want to use DIA-NN to run library search, it shows C:\diann\src\diann.cpp: 2749: unknown modification: 399.19418
ERROR: algorithmic failure
ERROR: ERROR: algorithmic failure

I tried to type "--strip-unknown-mods" in the additional option. It still shows the same error message. 

Would you please help me to fix it?

Thanks

**Comment by vdemichev** [comment]
Need to declare the modification with --mod in DIA-NN's options. Not sure why FragPipe does not do it automatically.

**Reply by grazzilybear** [reply]
Thanks. So if the modification is like this in the library C[399.19418]VNLVQK
I should declare the modification like this or which kind of information I should type? Thanks
--mod C[399.19418],399.19418,label

**Reply by vdemichev** [reply]
--mod 399.19418,399.19418

**Reply by grazzilybear** [reply]
It works!! Thanks for your great help!

**Reply by pejota66** [reply]
Hello

I have a similar issue  C:\diann\src\diann.cpp: 2749: unknown modification: Oxidation (P) when using the Mouse library from SwathAtlas. 

I have tried to use --mod but does not work. Could you please help me how to add "--mod Oxidation (P)" into the additional options?

Many thanks.
